recyclerview/recyclerview/src/main/java/androidx/recyclerview/widget/RecyclerView.java

```diff
@@ -13255,7 +13255,11 @@ public void onAnimationFinished(ViewHolder item) {
             // recycled.
             item.mShadowingHolder = null;
             if (!item.shouldBeKeptAsChild()) {
-                if (!removeAnimatingView(item.itemView) && item.isTmpDetached()) {
+                if (item.isTmpDetached()) {
+                    removeDetachedView(item.itemView, false);
+                    return;
+                }
+                if (!removeAnimatingView(item.itemView)) {
                     removeDetachedView(item.itemView, false);
```

Comment:
Is there any update on this PR? It would be nice to have an official fix for this. This GitHub PR _would be_ and official fix right?

Also would combining the `if` clauses achieve the same thing? Unless we're trying to be really explicit about the order.

```suggestion
                if (item.isTmpDetached() || !removeAnimatingView(item.itemView)) {
                    removeDetachedView(item.itemView, false);
                }
```

Reply:
it won't since the `removeAnimatingView` would throw the same `IllegalArgumentException`, since the item is temporarily detached